crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -628,16 +633,32 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
 
         let customer_id = payment_data.payment_intent.customer_id.clone();
 
+        let raw_customer_details = customer.map(|customer| CustomerData {
+            name: customer.name.map(|name| name.into_inner()).clone(),
+            email: customer.email.map(Email::from).clone(),
+            phone: customer.phone.map(|phone| phone.into_inner()).clone(),
+            phone_country_code: customer.phone_country_code.clone(),
+        });
+
+        // Updation of Customer Details for the cases where both customer_id and specific customer
+        // details are provided in Payment Create Request
```

Comment:
will this also handle updation of guest customer details? 
Consider a case where for a guest customer ( customer_id is not present ), name is passed in payments_create, and then in the confirm / update call, email is passed. Will the email be updated in payment intent?